### PR TITLE
Add line filename and line number information to acorn-optimizer errors

### PR DESCRIPTION
**Pull request by sbc100**
This helps a lot when trying to debug #20818.

### CODE REVIEW

File: tools/acorn-optimizer.js [hunk 1]
```diff
@@ -190,7 +199,7 @@ function restoreForVars(node) {
   let restored = 0;
   function fix(init) {
     if (init && init.type === 'EmptyStatement') {
-      assert(init.oldDeclarations);
+      assertAt(init.oldDeclarations, init.start);
```
Comment (kripken) [inline tools/acorn-optimizer.js:202]: Why isn't this
```suggestion
      assertAt(init.oldDeclarations, init);
```
? That is, I'd expect to see a node used and the location to mention as that same node, not a subnode of it.

File: tools/acorn-optimizer.js [hunk 2]
```diff
@@ -26,6 +26,15 @@ function assert(condition, text) {
   }
 }
 
+function assertAt(condition, offset, message) {
```
Comment (kripken) [inline tools/acorn-optimizer.js:29]:
```suggestion
function assertAt(condition, offset, message = '') {
```
(without this, won't not passing it in cause an "undefined" string to appear?

File: tools/acorn-optimizer.js (hunk 1 above)
Comment (sbc100) [reply tools/acorn-optimizer.js:202]: `start` and `end` are node attributes that refer to offsets within the input text.   i.e this is the offset of the node within the file, not a subnode.

**Comment by sbc100** [comment]
Writing a test is hard.. because these assertions all signal bugs in the optimizer itself, and not in the input.   If you could construct an input that would cause these assertions then that would be bug.. and we should fix that.

i.e. we don't have any known input that triggers these asserts.

File: tools/acorn-optimizer.js (hunk 1 above)
Comment (kripken) [reply tools/acorn-optimizer.js:202]: Oh, I see. Do we not always want the start? It seems simpler to pass in the node, and let `assertAt` take `.start` - we might also want to show the whole range actually, up to the end?

**Comment by kripken** [comment]
Makes sense, no need for a test then.

File: tools/acorn-optimizer.js (hunk 1 above)
Comment (sbc100) [reply tools/acorn-optimizer.js]: Done

**Review by kripken — APPROVED** [review]

**Pull request merged**